Hot fix lack of countries in WC 5.8 settings store

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Use `window.wcSettings.countries` as a fallback for Settings Store `getSetting( 'wc_admin', 'countries' )`, as the WC 5.8 does not have that field.

Hotfixes: https://github.com/woocommerce/google-listings-and-ads/issues/1041


This is by no means a complete fix, as we need more investigation results from https://github.com/woocommerce/woocommerce-admin/issues/7781 to know whether the backward-incompatible WC 5.8 change was desired or not. It would be nice to have tests for this hook, and the test that uses actual WC (see running tests with DEWP on multiple versions). Also, probably using globals is not recommended way to fetch settings for React components.

The idea is to fix it quickly.

### Screenshots:

![image](https://user-images.githubusercontent.com/17435/136825881-f266f625-af80-40fb-89bf-5bb469642b31.png)


### Detailed test instructions:

0. Install WC [5.8.0-rc.1](https://github.com/woocommerce/woocommerce/releases/tag/5.8.0-rc.1), or activate `main` branch of WC-admin.
1. If your MC is connected 
	1. got to Ads campaign setup [`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads).
	2. Go to the second step
2. If you MC is not set up, go to the audience selection step
3. See if countries are suggested for the Audience field
4. Do 1-3 with WC 5.7



<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Fix - Fetch countries for dropdowns in WC 5.8

tomalec committed:
Hot fix lack of countries in WC 5.8 settings store

Hot fixes: https://github.com/woocommerce/google-listings-and-ads/issues/1041

eason9487 reviewed (commented):
Tested with WC 5.7 and 5.8.0-RC1 versions and this fix works.

🚧 
But we have some similar issues to be addressed:

1. Another usage needs this patch as well
    https://github.com/woocommerce/google-listings-and-ads/blob/c21149d69e74300e12a2497e306865ab11df209c/js/src/hooks/useStoreCountry.js#L15
2. `.getSetting( 'wc_admin', 'adminUrl' )` is also not available and it could be fixed by the same way as `countries`
    https://github.com/woocommerce/google-listings-and-ads/blob/c21149d69e74300e12a2497e306865ab11df209c/js/src/hooks/useAdminUrl.js#L14-L17
3. `wcSettings.variationTitleAttributesSeparator` is not available and it will show a **undefined** wording on the product filter when searching variant products
    https://github.com/woocommerce/google-listings-and-ads/blob/c21149d69e74300e12a2497e306865ab11df209c/js/src/reports/products/async-requests.js#L17-L18
    ![image](https://user-images.githubusercontent.com/17420811/136925701-8b9a7ddb-27df-4f7c-a2aa-dd05b1da7e0b.png)
4. `wcSettings.woocommerceTranslation` is not available and the page title given by HTML source code will be changed -
    - from `Dashboard ‹ Google Listings & Ads ‹ Marketing ‹ WooCommerce ‹ wp.test — WooCommerce`
    - to `Dashboard ‹ Google Listings & Ads ‹ Marketing ‹  ‹ wp.test — WooCommerce`
    https://github.com/woocommerce/google-listings-and-ads/blob/c21149d69e74300e12a2497e306865ab11df209c/js/src/index.js#L27-L29

eason9487 commented:
> 3\. `wcSettings.variationTitleAttributesSeparator` is not available and it will show a **undefined** wording on the product filter when searching variant products
>     ![image](https://user-images.githubusercontent.com/17420811/136925701-8b9a7ddb-27df-4f7c-a2aa-dd05b1da7e0b.png)

Adding reproduce steps:

1. Go to the Reports Product page
2. Filter by the **Single Product** option with a variation product, such as `hoodie` imported from WC's sample data, and an additional filter will show up
3. Filter by the **Single Variation**

![Kapture 2021-10-12 at 17 13 13](https://user-images.githubusercontent.com/17420811/136928526-2b43a098-ac90-4343-9e97-aaf63284a59d.gif)

eason9487 commented:
Looks like `woocommerceTranslation` and `variationTitleAttributesSeparator` are moved under `wcSettings.admin` after WC 5.8.

findingsimple commented:
Wow nice catches @eason9487

tomalec committed:
Use `@woocommerce/wc-admin-settings` imported via DEWP,

instead of global `window.wcSettings` or settings store `getSetting`.

Note, that the package is not published and not added to dependencies.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1042#pullrequestreview-777087196

Works around https://github.com/woocommerce/woocommerce-admin/issues/7781

Co-authored-by: Eason Su <eason.su.tw@gmail.com>

tomalec commented:
I went with using `@woocommerce/wc-admin-settings` mapped to `wc.wcSettings` as suggested at https://github.com/woocommerce/google-listings-and-ads/issues/1041#issuecomment-940919492

It turned a number of hooks into simple utils. We could consider moving them to different folders, but I'd wait with it until we get a more official, future-proof solution for https://github.com/woocommerce/woocommerce-admin/issues/7781. Moving files obfuscates the git history, so I'd rather avoid moving files back and forth.

JPry commented:
I followed the test instructions, and can confirm that this PR works for both WooCommerce `5.8.0` and `5.7.1`. I also tested for the issue identified by @eason9487 and it looks like that has been resolved.

eason9487 reviewed (approved):
LGTM.
